curated_tokenizers/_bbpe.pyx

```diff
@@ -66,6 +66,14 @@ cdef class ByteBPEProcessor:
             c_merges.push_back(pair[string, string](p1.encode('utf-8'), p2.encode('utf-8')))
         self._merges = make_shared[Merges](c_merges)
 
+    def __copy__(self):
+        return ByteBPEProcessor(vocab=self.vocab, merges=self.merges)
+
+    def __deepcopy__(self, memo):
+        result = ByteBPEProcessor(vocab=self.vocab, merges=self.merges)
+        memo[id(self)] = result
```

Comment:
I think this violates the semantics of `deepcopy`:

> If the __deepcopy__() implementation needs to make a deep copy of a component, it should call the deepcopy() function with the component as first argument and the memo dictionary as second argument. The memo dictionary should be treated as an opaque object.

`deepcopy` itself fills memo. I think the `memo` argument is just there so that you can pass it through to recursive calls of `deepcopy` to preserve shared references.

Reply:
Removed the code that added the result to `memo`.